src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers/AttributeUpgradeAnalyzer.cs

```diff
@@ -41,6 +41,11 @@ public override void Initialize(AnalysisContext context)
 
             context.RegisterCompilationStartAction(context =>
             {
+                if (!context.Compilation.TargetsAspNetCore())
```

Comment:
Minimal - at the beginning of compilation it will check once, but that's it. If it's not targeting ASP.NET Core, it will not even run any syntax analyzers. So overall, for any user using the analyzer, there's probably a net gain performance over a solution.

Functionally, this won't put ASP.NET users in a weird place before they've moved over to .NET Core.